### ISSUE TITLE
Text arrays not escaped when serialized to text

### ISSUE THREAD

**Issue by marcoslot** [duckdb/duckdb#14463]
### What happens?

When a text array is serialized to text (e.g. during CSV write), the strings are not escaped, which means the serialized string cannot be coherently deserialized.

### To Reproduce

```sql
select ['a,c', 'b']::text as arr;
┌──────────┐
│   arr    │
│ varchar  │
├──────────┤
│ [a,c, b] │
└──────────┘

select len(['a,c', 'b']) as before, len(['a,c', 'b']::text::text[]) as after;
┌────────┬───────┐
│ before │ after │
│ int64  │ int64 │
├────────┼───────┤
│      2 │     3 │
└────────┴───────┘
```

### OS:

Ubuntu 22.04 x86_64

### DuckDB Version:

1.1.2

### DuckDB Client:

CLI

### Full Name:

Marco Slot

### Affiliation:

Crunchy Data

### What is the latest build you tested with? If possible, we recommend testing with the latest nightly build.

I have tested with a stable release

### Did you include all relevant data sets for reproducing the issue?

Yes

### Did you include all code required to reproduce the issue?

- [X] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [X] Yes, I have

**Comment by szarnyasg** [comment]
@marcoslot Thanks for reporting this! This may be implemented as a feature in the future but is not yet on the roadmap.

In the meantime, as a (crude) workaround, you can use `list_reduce`:
```sql
select '''' || list_reduce(['a,c', 'b'], (x, y) -> x || ''', ''' || y) || '''' as arr;
```
```text
┌────────────┐
│    arr     │
│  varchar   │
├────────────┤
│ 'a,c', 'b' │
└────────────┘
```

**Comment by marcoslot** [comment]
I think your list_reduce example also does not escape correctly, e.g.
```sql
select '''' || list_reduce(['a,''c', 'b'], (x, y) -> x || ''', ''' || y) || '''' as arr;
```
```
┌─────────────┐
│     arr     │
│   varchar   │
├─────────────┤
│ 'a,'c', 'b' │
└─────────────┘
```

fwiw, here's another way to reproduce:
```sql
copy (select ['hello, world'] arr) to '/tmp/test.csv';
select len(arr) from read_csv('/tmp/test.csv', columns={'arr':'TEXT[]'});
```
```
┌──────────┐
│ len(arr) │
│  int64   │
├──────────┤
│        2 │
└──────────┘
```

**Comment by taniabogatsch** [comment]
Indeed, we currently do not escape characters in `TEXT[] -> TEXT` casts, so the current workaround would be a custom escape function. Depending on your needs, you'd have to escape the delimiter, escape, and `[`, `]` characters...

The one below escapes commas.
```sql
list_transform(['a,c', 'b'], x -> replace(x, ',', ';'));
```

I agree that casts should round-trip, and probably be compatible with the default/sniffed CSV settings. 👍

**Comment by taniabogatsch** [comment]
Hi @marcoslot - this is addressed via https://github.com/duckdb/duckdb/pull/16304 and related PRs. :) And should be part of the next release.
